qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -411,32 +411,49 @@ def supports_aux_operators(cls) -> bool:
     def _eval_aux_ops(
         self,
         parameters: np.ndarray,
-        aux_operators: List[OperatorBase],
+        aux_operators: ListOrDict[OperatorBase],
         expectation: ExpectationBase,
         threshold: float = 1e-12,
     ) -> np.ndarray:
         # Create new CircuitSampler to avoid breaking existing one's caches.
         sampler = CircuitSampler(self.quantum_instance)
 
-        aux_op_meas = expectation.convert(StateFn(ListOp(aux_operators), is_measurement=True))
+        if isinstance(aux_operators, dict):
+            list_op = ListOp(list(aux_operators.values()))
+        else:
+            list_op = ListOp(aux_operators)
+
+        aux_op_meas = expectation.convert(StateFn(list_op, is_measurement=True))
         aux_op_expect = aux_op_meas.compose(CircuitStateFn(self.ansatz.bind_parameters(parameters)))
         values = np.real(sampler.convert(aux_op_expect).eval())
 
         # Discard values below threshold
         aux_op_results = values * (np.abs(values) > threshold)
-        # Deal with the aux_op behavior where there can be Nones or Zero qubit Paulis in the list
-        _aux_op_nones = [op is None for op in aux_operators]
-        aux_operator_eigenvalues = [
-            None if is_none else [result]
-            for (is_none, result) in zip(_aux_op_nones, aux_op_results)
-        ]
+
+        # Return None eigenvalues for None operators if aux_operators is a list.
+        # None operators are already dropped in compute_minimum_eigenvalue if aux_operators is a dict.
+        if isinstance(aux_operators, list):
+            aux_operator_eigenvalues = [None] * len(aux_operators)
+            key_value_iterator = enumerate(aux_op_results)
+        else:
+            aux_operator_eigenvalues = {}
+            key_value_iterator = zip(aux_operators.keys(), aux_op_results)
+
+        for key, value in key_value_iterator:
+            if aux_operators[key] is not None:
+                aux_operator_eigenvalues[key] = value
+        # # Deal with the aux_op behavior where there can be Nones or Zero qubit Paulis in the list
+        # aux_operator_eigenvalues = {
+        #     key: None if aux_operators[key] is None else result
+        #     for key, result in aux_op_results.items()
+        # }
```

Comment:
Why do you leave a copy of the old code here?

Reply:
I kept it as a reference while editing but forgot to remove it after I was finished. Will be removed in the next commit